### ISSUE TITLE
Reduce import time for `sunpy.map`

### ISSUE THREAD

**Issue by ayshih** [sunpy/sunpy#3448]
(I was getting annoyed by our import times, and was inspired by astropy/astropy#4598 to look deeper.)

Oof.  Importing `sunpy.map` takes a whopping 6+ seconds on my machine (see below).  There are a lot of contributing factors (see #3445, #3446, #3447), but what sticks out like a sore thumb is that 1.6 seconds (25% of the total time) is spent importing `sunpy.database` (which then imports `sunpy.net`) so that `map_factory` can handle `DatabaseEntry`.  Can that be re-factored so `map_factory` learns about `DatabaseEntry` only after a user has intentionally imported `sunpy.database`?

```
python -X importtime -c "import sunpy.map" 2> map.log
tuna map.log
```
![map](https://user-images.githubusercontent.com/991759/67576555-c6e3d900-f70c-11e9-9841-8ba77164ffd0.png)

**Comment by Cadair** [comment]
Also for `database` we only need the type to know how to read it, perhaps we can refactor `database` so we import less to know about the type.

**Comment by Cadair** [comment]
Closed by #3495??

**Comment by nabobalis** [comment]
Yes.
